src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceMultiSourceConfig.java

```diff
@@ -156,6 +157,43 @@ public void validate(FailureCollector collector) {
     return queries;
   }
 
+  /**
+   * validate whiteList and blackList SObjects
+   */
+  public void validateSObjects() {
+    DescribeGlobalResult describeGlobalResult;
+    try {
+      PartnerConnection partnerConnection =
```

Comment:
will this work when macros are enabled?

Reply:
yes it is working while macros is enabled.

Reply:
Can you list the tests that you executed?
What happens when auth credentials are specified as macros? Wouldn't `SalesforceConnectionUtil.getPartnerConnection(getAuthenticatorCredentials())` to fail in that case?

Reply:
we have added validations for macro for connection parameters.